File: Excel/676 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/676 Credit card payment amount.xlsx"
input = read_excel(path, range = "A1:B101")
test  = read_excel(path, range = "E1:G14")

result <- input %>%
  mutate(Date = as.Date(Date)) %>%
  mutate(adjusted_date = ifelse(day(Date) >= 26, Date + days(6), Date) %>% as.Date(origin = "1970-01-01")) %>%
  filter(year(adjusted_date) == 2025) %>%
  mutate(Quarter = paste0("Q", quarter(adjusted_date)),
         Month = month(adjusted_date, label = TRUE, abbr = TRUE, locale = "en")) %>%
  summarise(Payment = sum(Payment, na.rm = TRUE), .by = c("Quarter", "Month")) 

total = result %>%
  summarise(Payment = sum(Payment, na.rm = TRUE)) %>%
  mutate(Quarter = "Total", Month = NA) %>%
  relocate(Quarter, Month, Payment)

result = bind_rows(result, total)

File: Excel/600-699/676/676 Challenge.py
import pandas as pd
from pandas.tseries.offsets import DateOffset

path = "676 Credit card payment amount.xlsx"
input_df = pd.read_excel(path, usecols="A:B", nrows=101)
test_df = pd.read_excel(path, usecols="E:G", nrows=14)

input_df['Date'] = pd.to_datetime(input_df['Date'])
input_df['adjusted_date'] = input_df['Date'] + pd.to_timedelta((input_df['Date'].dt.day >= 26) * 6, unit='D')
input_df = input_df[input_df['adjusted_date'].dt.year == 2025]

result = (input_df
          .assign(Quarter="Q" + input_df['adjusted_date'].dt.quarter.astype(str),
                  Month=input_df['adjusted_date'].dt.strftime('%b'),
                  MonthOrder=input_df['adjusted_date'].dt.month)
          .groupby(['Quarter', 'Month', 'MonthOrder'], as_index=False)
          .agg(Payment=('Payment', 'sum'))
          .sort_values(by=['Quarter', 'MonthOrder'])
          .drop(columns=['MonthOrder']))

total = pd.DataFrame([{'Quarter': 'Total', 'Month': None, 'Payment': result['Payment'].sum()}])
result = pd.concat([result, total], ignore_index=True)
print(result)